test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/SelectTagHelperTest.cs

```diff
@@ -356,6 +356,91 @@ public class SelectTagHelperTest
             Assert.Equal(savedValue, items.Select(item => item.Value));
         }
 
+        [Fact]
+        public async Task ProcessAsync_WithItems_AndNoModelExpression_GeneratesExpectedOutput()
+        {
+            // Arrange
+            var originalAttributes = new TagHelperAttributeList
+            {
+                { "class", "form-control" },
+            };
+            var originalPostContent = "original content";
+
+            var expectedAttributes = new TagHelperAttributeList(originalAttributes);
+            var selectItems = new SelectList(Enumerable.Range(0, 5));
+            var expectedOptions = string.Join(
+                Environment.NewLine,
+                selectItems.Select(i => $"<option>HtmlEncode[[{i.Text}]]</option>")) + Environment.NewLine;
+
+            var expectedPreContent = "original pre-content";
+            var expectedContent = "original content";
+            var expectedPostContent = originalPostContent + expectedOptions;
+            var expectedTagName = "select";
+
+            var tagHelperContext = new TagHelperContext(
+                allAttributes: new TagHelperAttributeList(
+                    Enumerable.Empty<TagHelperAttribute>()),
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+            var output = new TagHelperOutput(
+                expectedTagName,
+                originalAttributes,
+                getChildContentAsync: (useCachedResult, encoder) =>
+                {
+                    var tagHelperContent = new DefaultTagHelperContent();
+                    tagHelperContent.AppendHtml("Something");
+                    return Task.FromResult<TagHelperContent>(tagHelperContent);
+                })
+            {
+                TagMode = TagMode.SelfClosing,
+            };
+            output.PreContent.AppendHtml(expectedPreContent);
+            output.Content.AppendHtml(expectedContent);
+            output.PostContent.AppendHtml(originalPostContent);
+
+            var metadataProvider = new TestModelMetadataProvider();
+            var htmlGenerator = new TestableHtmlGenerator(metadataProvider);
+            var viewContext = TestableHtmlGenerator.GetViewContext(
+                model: null,
+                htmlGenerator: htmlGenerator,
+                metadataProvider: metadataProvider);
+
+            var savedDisabled = selectItems.Select(item => item.Disabled).ToList();
+            var savedGroup = selectItems.Select(item => item.Group).ToList();
+            var savedSelected = selectItems.Select(item => item.Selected).ToList();
+            var savedText = selectItems.Select(item => item.Text).ToList();
+            var savedValue = selectItems.Select(item => item.Value).ToList();
```

Comment:
Drop the stuff confirming the user's `selectList` isn't changed. That's confirmed often enough in this class and this test is rather long.